examples/audio/eeg_signal.py

```diff
@@ -0,0 +1,554 @@
+"""
+Title: Electroencephalogram Signal Classification for action identification
+Author: [Suvaditya Mukherjee](https://github.com/suvadityamuk)
+Date created: 2022/11/03
+Last modified: 2022/11/05
+Description: Training a Convolutional model to classify EEG signals produced by exposure to certain stimuli.
+"""
+
+"""
+## Introduction
+
+The following example explores how we can make a Convolution-based Neural Network to
+perform classification on Electroencephalogram signals captured when subjects were
+exposed to different stimuli.
+We train a model from scratch since such signal-classification models are fairly scarce
+in pre-trained format.
+The data we use is sourced from the UC Berkeley-Biosense Lab where the data was collected
+from 15 subjects at the same time.
+Our process is as follows:
+
+- Load the [UC Berkeley-Biosense Synchronized Brainwave
+Dataset](https://www.kaggle.com/datasets/berkeley-biosense/synchronized-brainwave-dataset)
+Dataset](https://www.kaggle.com/datasets/berkeley-biosense/synchronized-brainwave-dataset)
+- Visualize random samples from the data
+- Pre-process, collate and scale the data to finally make a `tf.data.Dataset`
+- Prepare class weights in order to tackle major imbalances
+- Create a Conv1D and Dense-based model to perform classification
+- Define callbacks and hyperparameters
+- Train the model
+- Plot metrics from History and perform evaluation
+
+This example needs the following external dependencies (Gdown, Scikit-learn, Pandas,
+Numpy, Matplotlib). You can install it via the following commands.
+
+Gdown is an external package used to download large files from Google Drive. To know
+more, you can refer to its [PyPi page here](https://pypi.org/project/gdown)
+"""
+
+"""
+## Setup and Import headers
+"""
+
+"""shell
+pip install gdown
+pip install gdown
+pip install sklearn
+pip install pandas
+pip install numpy
+pip install matplotlib
```

Comment:
lets separate installs and add the following:

```python
"""
## Setup and Data Downloads

First, lets install our dependencies:
"""

"""shell
pip install gdown
pip install gdown
pip install sklearn
pip install pandas
pip install numpy
pip install matplotlib
"""

"""
Next, lets download our dataset.  
The gdown package makes it easy to download the data from Google drive:
"""

"""
gdown 1V5B7Bt6aJm0UHbR7cRKBEK8jx7lYPVuX
# gdown will download eeg-data.csv onto the local drive for use. Total size of
# eeg-data.csv is 105.7 MB
"""
```

Reply:
Done!